clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -10120,9 +10120,13 @@ def err_conflicting_sycl_kernel_attributes : Error<
 def err_sycl_attibute_cannot_be_applied_here
     : Error<"%0 attribute cannot be applied to a "
             "%select{static function or function in an anonymous namespace"
-            "|class member function"
-            "|function with a raw pointer return type"
-            "|function with a raw pointer parameter type}1">;
+            "|class member function}1">;
+def warn_sycl_attibute_address_space_optimization
+    : Warning<"%0 attribute applied to a "
+              "%select{function with a raw pointer return type"
+              "|function with a raw pointer parameter type}1"
+              " may prevent optimizations related to address spaces">,
```

Comment:
I'm not sure that "may prevent optimizations related to address spaces" is useful information. How about telling the truth: it should not really matter for our implementation, but it's not following the current SYCL spec. 
Other option is to guard this behavior under some command line option, see intel/llvm#806 .

Reply:
How about "SYCL 1.2.1 specification does not allow %0 attribute applied to a %select.."?

Reply:
Yeah, something like this... But to be honest, I'm not very good at wording.

Reply:
Done.